api-s3-datatypes.go

```diff
@@ -42,7 +42,7 @@ type commonPrefix struct {
 }
 
 // listBucketResult container for listObjects V2 response.
-type listBucketV2Result struct {
+type ListBucketV2Result struct {
 	// A response can contain CommonPrefixes only if you have
 	// specified a delimiter.
 	CommonPrefixes []commonPrefix
```

Comment:
I don't think this is a problem. Users won't declare commonPrefix themselves (as they could do with ListBucketV2Result) and Go won't complain about:

```prefixes := result.CommonPrefixes```